### PR TITLE
Feature/graphics

### PR DESCRIPTION
**Pull request by nlg8111**

### CODE REVIEW

File: src/components/Map.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, {Component} from "react"
+import {View} from "react-native"
+import styles from "../styles"
+
+export default class SizeAwareComponent extends Component {
```
Comment (PekkaAstala) [inline src/components/Map.js]: Something weird going on here? Every other class is defined in a file with the same name. I'd just say "Rename this class to Map?" but the implementation does seem to match this name more than a map.

File: src/components/MapRoadSingleLane.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const margins = 35
+    const centerY = this.height / 2
+    const verticalOffset = 30
+    return {
+      x: progress * (this.width - margins * 2) + margins,
+      y: centerY + verticalOffset
+    }
+  }
+
+  renderMapContent() {
```
Comment (PekkaAstala) [inline src/components/MapRoadSingleLane.js]: Render map content... As opposed to what content?

**Review by PekkaAstala — COMMENTED** [review]
I like the UnitContainer -> Line change and the "We have a Map that contains lines" change but the MapRoadSingleLane feels a bit weird. Maybe trying to make things too generic now that we only have one map and one line.

Maybe
Line: Good as it is
Map: Rename to SizeAwareComponent
MapRoadSingleLane: Rename to Map
?

File: src/components/Map.js (hunk 1 above)
Comment (nlg8111) [reply src/components/Map.js]: Yea, looks like my auto refactoring tools didn't rename it properly

File: src/components/MapRoadSingleLane.js (hunk 2 above)
Comment (nlg8111) [reply src/components/MapRoadSingleLane.js]: This is related to the whole map / sizeaware thing that is a bit wonky and goes against the grain of react, but maybe we can talk about it through VOIP.

**Pull request merged**